### ISSUE TITLE
[Deprecation] remove storage getters in FRAME

### ISSUE THREAD

**Issue by kianenigma** [paritytech/polkadot-sdk#223]
... or for now mark them as deprecated, and just remove all usage in substrate.

**Comment by kianenigma** [comment]
fwiw I think this is still worth doing, and worthy of a large tip. But it will be a hard PR to review, so one should brace themselves for a lot of merge conflicts :p

**Comment by kianenigma** [comment]
Happy to work on this in a step by step fashion, as described above.

**Comment by muraca** [comment]
I just opened a new issue to track the work in all the various pallets.  
According to the [deprecation checklist](https://github.com/paritytech/polkadot-sdk/blob/master/docs/contributor/DEPRECATION_CHECKLIST.md) I should add the deprecation message first. Is "`pallet::getter` will be removed after August 2024. Please use `StorageItem::get` instead." good? Should the PR for the deprecation message close this issue?

**Comment by ggwpez** [comment]
If you mark something as deprecated, then you also have to remove all usage - otherwise the CI will complain.

**Comment by muraca** [comment]
I'm fine with not following the deprecation checklist, due to the wide impact of this change

**Comment by ggwpez** [comment]
Yea we can just start with our pallets one-by-one or doing rarely modified ones first and then adding the deprecation once we removed it here.

**Comment by ggwpez** [comment]
Am not so convinced of this anymore. We are doing breaking changes to all our pallets for no obvious gain.

**Comment by xlc** [comment]
> Am not so convinced of this anymore. We are doing breaking changes to all our pallets for no obvious gain.

100%

we can do this is there are no other work left, which is clearly not the case. there are many CRITICAL open issues and people choose to break code and add more workload to everyone else instead

**Comment by shawntabrizi** [comment]
- There is nothing breaking about these changes if done right.
    - Just as a `pub fn get_variable` in `impl Pallet`
- It prevents more macro magic and needing to explain people how frame does stuff
- It prevents confusion for many people in the contract space which think these getters are needed to "read storage"
- It is a great open issue for brand new contributors to make their first pr
    - so it think it is not in conflict at all with people who actually work on critical open issues

**Comment by xlc** [comment]
yeah sure except people are making breaking changes and the breaking changes are merged

**Comment by shawntabrizi** [comment]
Sounds like a problem of developers and reviewers. Let's point out those problems, not point fingers at a good initiative.

**Comment by kianenigma** [comment]
Going forward, we should merge these without breaking change. The new CI warning you about it being a breaking change is enough deterrant to not want to break things needlessly.

There is no one working on this who could have solved anything more important. This initiaitve is a simple one for juniors to get their first contributions into Polkadot, and our support and review of it is value. Putting a stop gap on this is also not going to solve anything. I spend a few minutes every week reviewing this, and other should do the same.   

In fact, the same CI system and semver-based release is one of those CRITICAL things that we have indeed solved ;)
